src/server.c

```diff
@@ -58,7 +58,6 @@ void print_help() {
                     "    --uid, -u               User id to run with\n"
                     "    --gid, -g               Group id to run with\n"
                     "    --signal, -s            Signal to send to the command when exit it (default: SIGHUP)\n"
-                    "    --signal-list           Print a list of supported signals\n"
```

Comment:
useless and non working, once again doing non ttyd related work in ttyd.

UNIX is do one thing and do it well, listing signal is not a program function, nor a program option.

-1 is also non standard management of option 

and it s more like debugging code, not serious code, just like the optind torture , the server.c. look like a test program for corner case of getoptlong more than a lean way to load the program.

For a core feature like providing remote shell , features must be consider against the complexity of the program.

I tried to be kind of the massive removal of feature.

For example the all SSL code could be dropped too, IMHO:
want SSL ? put a proxy in front of it do not do it there.

Also the --gid and --uid are not the way to do in a sane, well integrated unix environment.

You should run su -l wathever -s ttyd or create a user _ttyd and always drop privilege to _ttyd, just like many program out there.

Reading about privsep https://www.openbsd.org/papers/eurobsdcon2016-privsep.pdf is a example of good thinking in an unix environment